### PR TITLE
UI Prototype

### PR DESCRIPTION
**Pull request by James-LG**
Wanted to use a native GUI, but CyberBattleSim uses plotly for its network graphs, and that is a Javascript library (browser support only). Using a very simple Flask web application instead.

image
image

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,135 @@
+ansiwrap==0.8.4
```
Comment (Christian-Fisher) [inline requirements.txt:1]: Should docker be looking for this file? (To auto install all the packages). If so, it didnt grab flask. I tried installing manually, but after doing that flask run doesnt actually do anything

File: web.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
+import json
+
+import plotly
+import plotly.graph_objs as go
+
+import cyberbattle.simulation.model as model
+import cyberbattle.simulation.commandcontrol as commandcontrol
+import cyberbattle.samples.toyctf.toy_ctf as ctf
+
+from flask import Flask, render_template
```
Comment (Christian-Fisher) [inline web.py:10]: Above

File: web.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+def generate_graph_json(dbg):
+    fig = go.Figure()
+    traces, layout = dbg.network_as_plotly_traces()
+    for t in traces:
+        fig.add_trace(t)
+    fig.update_layout(layout)
+
+    graph_json = json.dumps(fig, cls=plotly.utils.PlotlyJSONEncoder)
+    return graph_json
```
Comment (Christian-Fisher) [inline web.py:22]: So the graph can be serialized into a JSON? Thats very neat

**Review by Chrisnguyen3 — APPROVED** [review]
Looking pog

**Pull request merged**